src/index.ts

```diff
@@ -30,18 +30,19 @@ const getOptionalInput = (name: string) => core.getInput(name) || undefined;
   let { changesets } = await readChangesetState();
 
   let publishScript = core.getInput("publish");
-  let hasChangesets = changesets.length !== 0;
+  let shouldVersion = changesets.length !== 0;
+  let shouldPublish = changesets.length === 0;
   let hasPublishScript = !!publishScript;
 
   core.setOutput("published", "false");
   core.setOutput("publishedPackages", "[]");
-  core.setOutput("hasChangesets", String(hasChangesets));
```

Comment:
isnt this just changing the name? the functionality you are looking for is already here

Reply:
I took some liberties which I can understand reverting. The only change that really needs to be made is adding the variable to the actions.yml

Reply:
Ye, i would prefer it if u could bring back the previous name since renaming that would be a breaking change for me 😅 

https://github.com/statelyai/xstate-viz/blob/259fb02e170017139cc735f87e673e2125dcfb06/.github/workflows/version-or-release.yml#L39

Reply:
Interesting. So you're saying it works for you already? Wonder why it wasn't working for me.

I can just add the readme change and revert back to hasChangesets. Probably won't be until tomorrow. Thanks for the feedback!

Reply:
> So you're saying it works for you already?

I hope so 🤣 

> I can just add the readme change and revert back to hasChangesets. Probably won't be until tomorrow. Thanks for the feedback!

No worries, just leave edits to README and action.yml

Reply:
Updated